experimental/Schemes/ProjectiveSchemes.jl

```diff
@@ -1014,3 +1014,29 @@ of ``X`` and ``Y``, respectively.
   ff = CoveredSchemeMorphism(X, Y, phi)
   return ff
 end
+
+# Basic functionality required for Warham
+function dim(P::AbsProjectiveScheme{<:Field})
+  return dim(defining_ideal(P))-1
+end
+
+@attr function affine_algebra(P::AbsProjectiveScheme)
+  return quo(ambient_coordinate_ring(P), defining_ideal(P))[1]
+end
+
+@attr function hilbert_polynomial(P::AbsProjectiveScheme{<:Field})
+  return hilbert_polynomial(affine_algebra(P))
+end
+
+function degree(P::AbsProjectiveScheme{<:Field})
+  return degree(affine_algebra(P))
+end
+
+function arithmetic_genus(P::AbsProjectiveScheme{<:Field})
+  h = hilbert_polynomial(P)
+  return (-1)^dim(P) * (first(coefficients(h)) - 1)
+end
+
+@attr function is_smooth(P::AbsProjectiveScheme)
```

Comment:
Ideally specify the return type to allow calling code to be type stable
```suggestion
@attr Bool function is_smooth(P::AbsProjectiveScheme)
```

Reply:
Also holds for the other `@attr` in here. But this is not a blocker, just a quick observation.